File: typescript/tssearch/tests/searchfile_test.ts
/*
 * searchfile_test.js
 *
 * Some nodeunit tests of searchfile.js
 */

import {FileType} from '../src/filetype';
import {SearchFile} from '../src/searchfile';

exports.testSearchFileAbsPath = function(test) {
    const pathname = '/Users/cary/src/xsearch/javascript/jssearch/src';
    const filename = 'searchfile.js';
    const searchfile = new SearchFile(pathname, filename, FileType.Code);
    const expected = '/Users/cary/src/xsearch/javascript/jssearch/src/searchfile.js';
    test.ok(searchfile.relativePath() === expected, "searchfile path doesn't match expected");
    test.done();
};

exports.testSearchFileRelPath1 = function(test) {
    const pathname = '.';
    const filename = 'searchfile.js';
    const searchfile = new SearchFile(pathname, filename, FileType.Code);
    const expected = './searchfile.js';
    test.ok(searchfile.relativePath() === expected, "searchfile path doesn't match expected");
    test.done();
};

exports.testSearchFileRelPath2 = function(test) {
    const pathname = './';
    const filename = 'searchfile.js';
    const searchfile = new SearchFile(pathname, filename, FileType.Code);
    const expected = './searchfile.js';
    test.ok(searchfile.relativePath() === expected, "searchfile path doesn't match expected");
    test.done();
};

exports.testSearchFileRelPath3 = function(test) {
    const pathname = '..';
    const filename = 'searchfile.js';
    const searchfile = new SearchFile(pathname, filename, FileType.Code);
    const expected = '../searchfile.js';
    test.ok(searchfile.relativePath() === expected, "searchfile path doesn't match expected");
    test.done();
};

File: dart/dartsearch/test/search_file_test.dart
import 'dart:io' show File;

import 'package:test/test.dart';

void main() {
  test('test SearchFile with absolute path', () {
    var path = '/Users/cary/src/xsearch/dart/dartsearch/lib/src/search_file.dart';
    var searchFile = SearchFile(File(path), FileType.code);
    expect(searchFile.toString(), path);
  });

  test('test SearchFile with tilde path', () {
    var path = '~/src/xsearch/dart/dartsearch/lib/src/search_file.dart';
    var searchFile = SearchFile(File(path), FileType.code);
    expect(searchFile.toString(), path);
  });

  test('test SearchFile with rel path #1', () {
    var path = './search_file.dart';
    var searchFile = SearchFile(File(path), FileType.code);
    expect(searchFile.toString(), path);
  });

  test('test SearchFile with rel path #2', () {
    var path = './search_file.dart';
    var searchFile = SearchFile(File(path), FileType.code);
    expect(searchFile.toString(), path);
  });
}
